### ISSUE TITLE
Namespace transformer ID conflict error message is useless

### ISSUE THREAD

**Issue by thefirstofthe300** [argoproj/argo-cd#3111]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: http://bit.ly/argocd-faq.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When Argo attempts to sync one of our applications, it throws the error:

```
ComparisonError: rpc error: code = Unknown desc = `kustomize build /tmp/git@<snip>/metrics-server` failed exit status 1: Error: namespace tranformation produces ID conflict: []*resource.Resource{(*resource.Resource)(0xc0002d5ea0), (*resource.Resource)(0xc0001d0140)}
```

This makes it very difficult to figure out what resources are colliding, especially when the configuration deploys from my laptop just fine without any conflicts.

**Expected behavior**

I would expect the error message to be much more clear than what it is. The issue would be in the underlying kustomize build and this particular issue appears to have been fixed in [kustomize v3.5.4 here](https://github.com/kubernetes-sigs/kustomize/pull/2012). Can we update kustomize to this version? I'm running a custom container because we needed the aws authenticator v0.5.0 which wasn't released until recently. We're planning to move back to standard containers anyway and this would give us a good excuse to get this updated.

**Version**

```shell
➜ argocd version                                       
argocd: v1.3.6+89be1c9
  BuildDate: 2019-12-10T22:48:19Z
  GitCommit: 89be1c9ce6db0f727c81277c1cfdfb1e385bf248
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v1.4.0+2d02948
  BuildDate: 2020-01-18T05:55:02Z
  GitCommit: 2d029488aba6e5ad48b2a756bfcf43d5cb7abcee
  GitTreeState: clean
  GoVersion: go1.12.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: Version: {Version:kustomize/v3.2.1 GitCommit:d89b448c745937f0cf1936162f26a5aac688f840 BuildDate:2019-09-27T00:10:52Z GoOs:linux GoArch:amd64}
  Helm Version: v2.15.2
  Kubectl Version: v1.14.0
```

**Comment by thefirstofthe300** [comment]
I was able to track down the underlying cause of the issue here. I downloaded v3.5.4 of kustomize, attempted to run the build, parsed the error message, and properly deduplicated the configuration.

The version of kustomize bundled with kubectl appears to be old enough that it doesn't check for duplicate resources and consequently, doesn't error at all.

Still, the extra information provided by v3.5.4 of kustomize was very helpful in tracking down the issue here.
